src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java

```diff
@@ -660,20 +663,15 @@ private void handleFinished()
                 correlations.put(newCorrelationId, handshake);
 
                 final long newApplicationId = supplyStreamId.getAsLong();
-                final long applicationRef = route.targetRef();
 
-                String applicationProtocol = tlsEngine.getApplicationProtocol();
-                if (applicationProtocol == null || applicationProtocol.isEmpty())
-                {
-                    applicationProtocol = tlsRouteEx.applicationProtocol().asString();
-                }
-                if (applicationProtocol == null || applicationProtocol.isEmpty())
+                String tlsApplicationProtocol = tlsEngine.getApplicationProtocol();
+                if (tlsApplicationProtocol == null || tlsApplicationProtocol.isEmpty())
```

Comment:
Only test for `!= null` since empty string is a valid value for application protocol.

Reply:
SSLEngine#getApplicationProtocol() returns "". From javadoc,

"null if it has not yet been determined if application protocols might be used for this connection, an empty String if application protocols values will not be used, or a non-empty application protocol String if a value was successfully negotiated."

Reply:
good clarification - thanks, let's revert `""` back to `null` to it appears "unset" for routing purposes.